src/OrbitBase/include/OrbitBase/File.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_FILE_H_
+#define ORBIT_BASE_FILE_H_
+
+#include <absl/strings/str_format.h>
+#include <fcntl.h>
+
+#include <filesystem>
+#include <string_view>
+
+#include "OrbitBase/Result.h"
+#include "OrbitBase/SafeStrerror.h"
+#include "OrbitBase/UniqueResource.h"
+
+#if defined(__linux)
+#include <unistd.h>
+#elif defined(_WIN32)
+#include <io.h>
+// Windows does not have TEMP_FAILURE_RETRY - define a shortcut
+#define TEMP_FAILURE_RETRY(expression) (expression)
+#endif
```

Comment:
Could this be moved to the CPP file? Or is there still a reason for the user to use TEMP_FAILURE_RETRY?

Reply:
It is currently used in ReadFileToString.cpp